### PR TITLE
docs: document metrics-generator ring

### PR DESCRIPTION
**Pull request by kvrhdn**
**What this PR does**:
Extend the API reference and the consistent hash ring page to list the new metrics-generator ring.

This is the first (of many) PRs to document the metrics-generator. We will also have to describe the new config options in the configuration section and update various other pages.

**Which issue(s) this PR fixes**:
Related to https://github.com/grafana/tempo/issues/1303

**Checklist**
- [ ] ~~Tests updated~~
- [x] Documentation added
- [ ] ~~`CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`~~

### CODE REVIEW

**Review by joe-elliott — APPROVED** [review]

File: docs/tempo/website/api_docs/_index.md [hunk 1]
```diff
@@ -299,7 +300,17 @@ Displays a web page with the ingesters hash ring status, including the state, he
...
 
+### Metrics-generators ring status
 
+> Note: this endpoint is only available when the metrics-generator is enabled. See [configuration](../configuration/_index.md#metrics-generator).
+
+```
+GET /metrics-generator/ring
+```
+
+Displays a web page with the metrics-generators hash ring status, including the state, healthy and last heartbeat time of each metrics-generator.
```
Comment (KMiller-Grafana) [inline docs/tempo/website/api_docs/_index.md:311]:
```suggestion
Displays a web page with the metrics-generator hash ring status, including the state, health, and last heartbeat time of each metrics-generator.

This endpoint is only available when the metrics-generator is enabled. See [metrics-generator](../configuration/_index.md#metrics-generator).
```

File: docs/tempo/website/operations/consistent_hash_ring.md [hunk 2]
```diff
@@ -48,6 +54,12 @@ Unhealthy distributors have little impact but should be forgotten to reduce cost
 
 Unhealthy ingesters will cause writes to fail. If the ingester is really gone forget immediately!
 
+#### Metrics-generators
+**Available on:** Distributors  
+**Path:** `/metrics-generator/ring`
+
+Unhealthy metrics-generators will cause writes to fail. If the metrics-generator is really gone forget immediately!
```
Comment (KMiller-Grafana) [inline docs/tempo/website/operations/consistent_hash_ring.md:61]: We don't use exclamation points in documentation.  Even if we really want to!

```suggestion
Unhealthy metrics-generators will cause writes to fail. If the metrics-generator is really gone, forget it immediately.
```

**Review by KMiller-Grafana — DISMISSED** [review]

**Comment by kvrhdn** [comment]
Thank you for the suggestions @KMiller-Grafana. Should I apply the same changes to the other sections on these pages?
Every ring has more or less the same information, but the new ring will have improved wording now.

File: docs/tempo/website/operations/consistent_hash_ring.md [hunk 3]
```diff
@@ -25,6 +25,12 @@ This ring is only used when "global" rate limits are used. The distributors use
 
 This ring is used by the distributors to load balance traffic into the ingesters. When spans are received the trace id is hashed and they are sent to the appropriate ingesters based on token ownership in the ring. Queriers also use this ring to find the ingesters for querying recent traces.
 
+#### Metrics-generator
+**Participants:** Metrics-generators  
+**Used by:** Distributors
+
+This ring is used by distributors to load balance traffic to the metrics-generators. When spans are received, the trace ID is hashed, and the traces are sent to the appropriate metrics-generators based on token ownership in the ring.
```
Comment (zalegrala) [inline docs/tempo/website/operations/consistent_hash_ring.md:32]: Should we mention what the hash is?

**Review by zalegrala — APPROVED** [review]

File: docs/tempo/website/operations/consistent_hash_ring.md (hunk 3 above)
Comment (kvrhdn) [reply docs/tempo/website/operations/consistent_hash_ring.md:32]: I think that's an implementation detail of the ring in dskit. How a hash ring works is explained here: https://cortexmetrics.io/docs/architecture/#the-hash-ring (this link is also on the top of this page in the docs)

**Pull request merged**